src/java.base/share/classes/java/lang/invoke/InnerClassLambdaMetafactory.java

```diff
@@ -642,7 +643,7 @@ boolean requiresPreload(Class<?> cls) {
             while (c.isArray()) {
                 c = c.getComponentType();
             }
-            return (c.isValue() && !PrimitiveClass.isPrimitiveClass(c)) || PrimitiveClass.isPrimitiveValueType(c);
```

Comment:
Perhaps there should be a method that computes this compound expression?

Reply:
This compound expression should not be needed.  Preload attribute is generated for a value class including primitive class.  It's a leftover when we added support for value objects.